File: Iverson.Clients/TypeScript/sample/models/Article.ts
import 'reflect-metadata';
import {
    IversonEntity,
    IversonGuid,
    IversonKey,
    IversonLargeField,
    IversonSearchKey,
    ManyToOne,
} from '../../src/annotations.js';
import { Author } from './Author.js';

@IversonEntity()
export class Article {
    @IversonKey()
    @IversonGuid()
    id: string = '';

    title: string = '';

    @IversonLargeField()
    body: string = '';

    @IversonSearchKey(0)
    category: string = '';

    wordCount: number = 0;

    @IversonSearchKey(1)
    publishedAt: Date = new Date();

    @ManyToOne(() => Author)
    authorId: string = '';
}

File: Iverson.Clients/Go/sample/models/article.go
package models

import "time"

// Article is a sample entity demonstrating all Iverson struct tag forms.
type Article struct {
	Id          string `iverson_key:"true" iverson_guid:"true"`
	Title       string
	Body        string `iverson_large_field:"true" iverson_chunk:"256:32"`
	Category    string `iverson_search_key:"0"`
	WordCount   int
	PublishedAt time.Time `iverson_search_key:"1"`
	AuthorId    string    `iverson:"many_to_one:Author"`
}
